apollo-router/src/request.rs

```diff
@@ -165,27 +165,101 @@ impl Request {
     ///
     /// An error will be produced in the event that the query string parameters
     /// cannot be turned into a valid GraphQL `Request`.
-    pub fn from_urlencoded_query(url_encoded_query: String) -> Result<Request, serde_json::Error> {
-        let urldecoded: serde_json::Value =
-            serde_urlencoded::from_bytes(url_encoded_query.as_bytes())
-                .map_err(serde_json::Error::custom)?;
+    pub(crate) fn batch_from_urlencoded_query(
+        url_encoded_query: String,
+    ) -> Result<Vec<Request>, serde_json::Error> {
+        let value: serde_json::Value = serde_urlencoded::from_bytes(url_encoded_query.as_bytes())
+            .map_err(serde_json::Error::custom)?;
 
-        let operation_name = if let Some(serde_json::Value::String(operation_name)) =
-            urldecoded.get("operationName")
-        {
-            Some(operation_name.clone())
+        Request::process_query_values(&value)
+    }
+
+    /// Convert Bytes into a GraphQL [`Request`].
+    ///
+    /// An error will be produced in the event that the query string parameters
+    /// cannot be turned into a valid GraphQL `Request`.
+    pub(crate) fn batch_from_bytes(bytes: &[u8]) -> Result<Vec<Request>, serde_json::Error> {
+        let value: serde_json::Value =
+            serde_json::from_slice(bytes).map_err(serde_json::Error::custom)?;
+
+        Request::process_batch_values(&value)
+    }
+
+    fn allocate_result_array(value: &serde_json::Value) -> Vec<Request> {
+        match value.as_array() {
+            Some(array) => Vec::with_capacity(array.len()),
+            None => Vec::with_capacity(1),
+        }
+    }
+
+    fn process_batch_values(value: &serde_json::Value) -> Result<Vec<Request>, serde_json::Error> {
+        let mut result = Request::allocate_result_array(value);
+
+        if value.is_array() {
+            tracing::info!(
+                histogram.apollo_router.operations.batching.size = result.len() as f64,
+                mode = "batch_http_link" // Only supported mode right now
```

Comment:
NIT: for these metrics you should implement Display on the enum Mode and just use `%BatchingMode::HttpLink` to avoid typo

Reply:
Yep, that's a hangover from when I first implemented it as a `String`. `Display` is implemented already, but I overlooked updating it.

done: 4fd50dd142795bee0c2ef053bc54050fc9d19fee